lib/src/sql/cast.rs

```diff
@@ -0,0 +1,87 @@
+use crate::ctx::Context;
+use crate::dbs::Options;
+use crate::dbs::Transaction;
+use crate::err::Error;
+use crate::sql::comment::mightbespace;
+use crate::sql::error::IResult;
+use crate::sql::idiom::Idiom;
+use crate::sql::kind::{kind, Kind};
+use crate::sql::value::{single, Value};
+use async_recursion::async_recursion;
+use nom::character::complete::char;
+use nom::sequence::delimited;
+use serde::{Deserialize, Serialize};
+use std::cmp::Ordering;
+use std::fmt;
+
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize, Hash)]
+#[serde(rename = "$surrealdb::private::sql::Cast")]
+pub struct Cast(pub Kind, pub Value);
```

Comment:
Code might be cleaner (less `.0` and `.1`) if the fields were named. Hopefully that wouldn't affect derived serialization in a bad way.

Reply:
Let's change this in a separate PR @finnbear 👍